pkg/compose/compose.go

```diff
@@ -216,9 +219,11 @@ func (s *composeService) projectFromName(containers Containers, projectName stri
 				}
 				service.DependsOn[dependency] = types.ServiceDependency{Condition: condition, Restart: restart, Required: required}
 			}
+			set[name] = service
```

Comment:
this might not be practically problematic, but in general modifying a map during it's iteration is bad practice and can produce unpredictable results

Reply:
an alternative could be to do this inside the other `for` (where we build `set`), which might also be slightly more efficient?

Reply:
> in general modifying a map during it's iteration is bad practice and can produce unpredictable results

any reference ?
I was looking for official reference regarding this and only found https://github.com/golang/go/issues/9926

Reply:
Ah, that's fair. I probably got mixed up with adding key/value pairs, but modifying is safe (which makes sense). It was lots of code to comb through 😅